src/cockpit/packages.py

```diff
@@ -194,11 +194,37 @@ def __init__(self, value: JsonObject):
 
 
 class Condition:
-    def __init__(self, value: JsonObject):
-        try:
-            (self.name, self.value), = value.items()
-        except ValueError as exc:
-            raise JsonError(value, 'must contain exactly one key/value pair') from exc
+    def get_condition_files(self) -> Iterable[str]:
+        raise NotImplementedError
+
+    def check(self, path_exists: Callable[[str], bool]) -> bool:
+        raise NotImplementedError
+
+
+class UnknownCondition(Condition):
+    def __init__(self, value: JsonObject, path: Path) -> None:
+        self.value = value
+        self.path = path
+
+    def get_condition_files(self) -> Iterable[str]:
+        return []
+
+    def check(self, path_exists: Callable[[str], bool]) -> bool:
+        # do *not* ignore manifests with unknown predicates, for forward compatibility
+        logger.warning('  %s: ignoring unknown predicate in manifest: %s', self.path, self.value)
+        return True
+
+
+class PathExistsCondition(Condition):
+    def __init__(self, path: str, *, exists: bool) -> None:
+        self.path = path
+        self.exists = exists
```

Comment:
Was wondering if you were going to do this as a unified thing at some point.  Makes sense :+1: